### ISSUE TITLE
Generate POJOs from XSD

### ISSUE THREAD

**Issue by VincentX66** [NeTEx-CEN/NeTEx#99]
Hi, guys,

has anyone managed to generate code from the xsd? I tried with xmlbeans and xjc (jaxb), but only get error messages (like that something is already in use or collisions). I've already looked for solutions, but unfortunately found none. It kept saying, "use the xsd to generate classes".

**Comment by skinkie** [comment]
@VincentX66 this is why you always should ask questions like this :) It is done, and currently maintained:

https://github.com/entur/netex-java-model

**Comment by VincentX66** [comment]
@skinkie Thank you for your quick answer and your correction of my question. Sorry, never used GitHub before. Anyway, https://github.com/entur/netex-java-model doesn't work for me. A problem with the maven-exec-plugin and that it's not a windows application. I don't know if I should go into more detail or note it in the appropriate github repository.

**Comment by skinkie** [comment]
I think it is better to ask there. But the command "mvn" should work on windows as well. Worst case you can obviously run this in Linux generate the classes ;-) 

Or use the precompiled code: https://mvnrepository.com/artifact/org.entur/netex-java-model/1.0.9

**Comment by janosu** [comment]
Hello Stefan,

I looked at your bindings.xjb and would have a comment for you. The xsd:duration is assigned to java.time.Duration, but java.time.Duration cannot handle the full format specified in the standard.
See: 
https://www.w3.org/TR/xmlschema-2/#duration
https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-

We use javax.xml.datatype.Duration.

**Comment by skinkie** [comment]
@janosu please inform entur about this in their repository.

**Comment by janosu** [comment]
Yes, it can be more than one day (eg.: for a train to Moscow), but it is not only in this field that the duration type can be. For example, in the case of tickets, season tickets, where the period of validity can be specified, which can be more than one day, eg: 3 months, half a year, etc.

So in general, in xsd: duration you can specify a year and a month in addition to the days and times that java.time.Duration can't handle.

**Comment by skinkie** [comment]
@janosu the current Duration type just encodes it to seconds and nanoseconds right? No further limitations? Or is the parsing an issue?

**Comment by janosu** [comment]
Yes,  just encodes it to seconds and nanoseconds. Look at the source code.
Another example: See https://www.baeldung.com/java-period-duration
The duration between dates is handled by java.time.Period.

The java.time.Duration can parse "PnDTnHnMn.nS", but W3C xs:duration can be "-PnYnMnDTnHnMn.nS".
See https://www.w3schools.com/xml/schema_dtypes_date.asp

**Comment by pieter90** [comment]
@skinkie Do you know whether the https://github.com/entur/netex-java-model project is still being maintained? If I'm correct there have been two Netex release (1.2.3 and 1.3.1) since the last release of netex-java-model.

I am trying to generate Java objects from the Netex xsd, so far unsuccessfully. Using netex-java-model would be convenient, but I'd like to use the latest Netex version.

**Comment by skinkie** [comment]
@pieter90 that should still be the basis for most of Entur's software. It is rather easy to create your own POJO's with it. If you want to do it for The Netherlands, you would have to anyway, considering that development and alignment happens in the next-branch.
